File: Common/Implementations/AppSettings.cs
using System;
using System.Collections.Generic;
using System.Text;
using Common.Interfaces;

namespace Common
{
    public class CommonSettings
    {
        public string DomainUrl { get; set; }
        public string ApiUrl { get; set; }
        public string ApiName { get; set; }
    }
    public class AuthSettings
    {
        public string Authority { get; set; }
        public string ClientId { get; set; }
        public string ClientSecret { get; set; }
        public string RedirectUri { get; set; }
        public string ResponseType { get; set; }
        public string Scope { get; set; }
    }

    public class AppSettings : IAppSettings 
    {
        public CommonSettings CommonSettings { get; set; } = new CommonSettings();
        public AuthSettings AuthSettings { get; set; } = new AuthSettings();
    }
}

File: UI/ClientApp/code/config/AppSettings.ts
import { UserManagerSettings } from 'oidc-client'

interface ICommonSettings {
    apiUrl: string
};

interface IAuthSettings {
    authority: string,
    clientId: string,
    scope: string 
    oidcConfig: UserManagerSettings
}

interface IAppSettings {
    common: ICommonSettings,
    auth: IAuthSettings,
   
}


const settings = (document as any)["app"]["settings"] as IAppSettings;

settings.auth.oidcConfig = <UserManagerSettings>{
    authority: settings.auth.authority,
    client_id: settings.auth.clientId,
    redirect_uri: `${window.location.protocol}//${window.location.hostname}:${window.location.port}/oidc-callback`,
    response_type: "id_token token",
    scope: "openid profile api",
    silent_redirect_uri: `${window.location.protocol}//${window.location.hostname}:${window.location.port}/oidc-callback`,
    popup_redirect_uri: `${window.location.protocol}//${window.location.hostname}:${window.location.port}/oidc-callback`,
    popup_post_logout_redirect_uri: `${window.location.protocol}//${window.location.hostname}:${window.location.port}/oidc-callback`,
    automaticSilentRenew: true,
    filterProtocolClaims: true,
    //loadUserInfo: true
}


export const appSettings = {
    ...settings,
    
} as IAppSettings